src/generic.jl

```diff
@@ -0,0 +1,56 @@
+function LinearAlgebra.lu(A::CuSparseMatrixCSR{T,Cint}) where T <: BlasFloat
+  n = LinearAlgebra.checksquare(A)
+  solver = CudssSolver(A, 'G', 'F')
+  x = CudssMatrix(T, n)
+  b = CudssMatrix(T, n)
+  cudss("analysis", solver, x, b)
+  cudss("factorization", solver, x, b)
+  return solver
+end
+
+function LinearAlgebra.ldlt(A::CuSparseMatrixCSR{T,Cint}) where T <: BlasFloat
+  n = LinearAlgebra.checksquare(A)
+  structure = T <: Real ? 'S' : 'H'
+  solver = CudssSolver(A, structure, 'F')
+  x = CudssMatrix(T, n)
+  b = CudssMatrix(T, n)
+  cudss("analysis", solver, x, b)
+  cudss("factorization", solver, x, b)
+  return solver
+end
+
+function LinearAlgebra.cholesky(A::CuSparseMatrixCSR{T,Cint}) where T <: BlasFloat
+  n = LinearAlgebra.checksquare(A)
+  structure = T <: Real ? "SDP" : "HDP"
+  solver = CudssSolver(A, structure, 'F')
+  x = CudssMatrix(T, n)
+  b = CudssMatrix(T, n)
+  cudss("analysis", solver, x, b)
+  cudss("factorization", solver, x, b)
+  return solver
+end
+
+for fun in (:lu!, :ldlt!, :cholesky!)
+  @eval begin
+    function LinearAlgebra.$fun(solver::CudssSolver, A::CuSparseMatrixCSR{T,Cint}) where T <: BlasFloat
+      n = LinearAlgebra.checksquare(A)
+      cudss_set(solver.matrix, A)
+      x = CudssMatrix(T, n)
+      b = CudssMatrix(T, n)
```

Comment:
Should we allocate two new vectors each time we are updating the factorization?

Reply:
Nop, I just create a dummy `CudssMatrix` for `x` and `b` with these new constructors [here](https://github.com/exanauts/CUDSS.jl/pull/7/files#diff-754e6441f2054c00a80b0953b1876f78a4f3e85993eebd6fa0e3b883611a58caR33-R51).
I use a similar trick with dummy descriptors for `KrylovOperator` and `TriangularOperator` in `KrylovPreconditioners.jl.